### PR TITLE
Arraycollection matching method multiple ordering fix

### PR DESCRIPTION
**Pull request by baileylo**
This is a copy of #59 that includes unit tests.

### CODE REVIEW

File: tests/Doctrine/Tests/Common/Collections/ArrayCollectionTest.php [hunk 1]
```diff
@@ -293,4 +293,26 @@ public function testMatchingWithSortingPreservesyKeys()
...
+
+    public function testMultiColumnSortAppliesAllSorts()
+    {
+        $collection = new ArrayCollection(array(
+            array('foo' => 1, 'bar' => 2),
+            array('foo' => 2, 'bar' => 4),
+            array('foo' => 2, 'bar' => 3)
+        ));
+
+        $expected = [
```
Comment (stof) [inline tests/Doctrine/Tests/Common/Collections/ArrayCollectionTest.php]: this is not compatible with PHP 5.3, which is still supported

Comment (baileylo) [inline tests/Doctrine/Tests/Common/Collections/ArrayCollectionTest.php]: Sorry, missed that short array syntax.

**Comment by baileylo** [comment]
@stof Removed short array syntax issues.

**Comment by stof** [comment]
:+1:

**Comment by stof** [comment]
@Ocramius it would be great to merge this bugfix and to release 1.3.1

**Comment by delasallejean** [comment]
@baileylo thanks !
Closing #59

**Comment by baileylo** [comment]
@stof Do you know where we are with merging this?

**Comment by fsevestre** [comment]
@stof Any news about this PR ?

**Comment by deeky666** [comment]
Thanks! Tagging is up to @Ocramius though.

**Comment by respinoza** [comment]
@Ocramius Any news on tagging a new release including this fix?

Thank you.

**Comment by zBart** [comment]
Also having this issue, a 1.3.1 update would be great!

**Comment by sdufel** [comment]
@Ocramius  : +1 for a 1.3.1

**Comment by respinoza** [comment]
@deeky666 Would be possible to tag a new release ? (1.3.1)
Thank you!

**Comment by jakeasmith** [comment]
Sooo... how 'bout that 1.3.1? Or is this project no longer maintained?

**Comment by guilhermeblanco** [comment]
We should tag a 1.4 version soon... =)
Just need to review/merge other non-BC breaking features... demands some time to carefully review and accept them. My current focus is setting the path for ORM v3, but I'm sure other contributors can handle that in a reasonable timeframe.

**Comment by Ocramius** [comment]
Likely going to work on these repos after DPC, which is at the end of the month.

**Comment by trickeyone** [comment]
Could we maybe get a new release including this?

**Comment by mikeSimonson** [comment]
@trickeyone Done

**Comment by deeky666** [comment]
@mikeSimonson what am I supposed to do here?

**Comment by mikeSimonson** [comment]
@deeky666 nothing, but you merge the PR so it was assigned to you. I taught that it was the new policy :tm:

**Comment by deeky666** [comment]
@mikeSimonson you are right I did indeed :D Didn't remember. Thanks.

**Pull request merged**